scripts/db-seeding/fillTestMessagesData.js

```diff
@@ -0,0 +1,151 @@
+          process.exit(1);
+        }
+
+        (function addNextMessageThread() {
+          var messageThread = new Thread;
+          var threadSize = random(maxMessages) + 1;
+
+          (function addNextMessage(depth, to, from) {
+            var message = new Message();
+
+            message.created = addDays(Date.now(), -depth + 1);
+            message.content = faker.lorem.sentences();
+
+            // Randomize indecies
+            var randomUsers = [];
+            for (var i = 0; i < users.length; i++) {
+              randomUsers[i] = i;
+            }
+            randomUsers = _.shuffle(randomUsers);
+
+            if (to) {
+              message.userTo = to;
+            } else {
+              message.userTo = users[randomUsers[1]]._id;
+            }
+            if (from) {
+              message.userFrom = from;
+            } else {
+              message.userFrom = users[randomUsers[0]]._id;
+            }
+
+            // Assume 80% of messages are read
+            if (random(100) < 80) {
+              message.read = true;
+            } else {
+              message.read = false;
+            }
+
+            message.notificationCount = 0;
+
+            message.save(function (err) {
+              if (err != null) {
+                console.log(err);
+              }
+            });
+
+            // Add thread for the most recent message
+            if (depth === 1) {
+              messageThread.updated = message.created;
+              messageThread.userFrom = message.userFrom;
+              messageThread.userTo = message.userTo;
+              messageThread.message = message._id;
+              messageThread.read = false;
+              messageThread.save(function (err) {
+                if (err != null) {
+                  console.log(err);
+                }
+                else {
+                  console.log('index ' + index);
+                  if (index >= numThreads) {
+                    console.log(chalk.green('Done with ' + numThreads + ' test threads!'));
+                    console.log(chalk.white('')); // Reset to white
+                    process.exit(0);
+                  }
+                }
+              });
+            }
+
+            depth-=1;
+            if (depth > 0) {
+              // Reverse the order of to and from to simulate a conversation going back and forth
+              addNextMessage(depth, message.userFrom, message.userTo);
+            }
+
+          }(threadSize));
+
+          index+=1;
+          if (index < numThreads) {
+            addNextMessageThread();
+          }
+        }());
+
+      }).catch(function (err) {
+        console.log(err);
+      });
+    });
+  });
+};
+
+// Number of threads and max number of messages is required
+if (process.argv[2] == null || process.argv[2] < 1 || process.argv[3]== null || process.argv[3] < 1) {
```

Comment:
As a future improvement, we could build this stuff using [Yargs](http://yargs.js.org/), but that's for another PR.